homeassistant/components/sensor/bme680.py

```diff
@@ -0,0 +1,373 @@
+DEFAULT_AQ_BURN_IN_TIME = 300  # 300 second burn in time for AQ gas measurement
+DEFAULT_AQ_HUM_BASELINE = 40  # 40%, an optimal indoor humidity.
+DEFAULT_AQ_HUM_WEIGHTING = 25  # 25% Weighting of humidity to gas in AQ score
+
+SENSOR_TEMP = 'temperature'
+SENSOR_HUMID = 'humidity'
+SENSOR_PRESS = 'pressure'
+SENSOR_GAS = 'gas'
+SENSOR_AQ = 'airquality'
+SENSOR_TYPES = {
+    SENSOR_TEMP: ['Temperature', None],
+    SENSOR_HUMID: ['Humidity', '%'],
+    SENSOR_PRESS: ['Pressure', 'mb'],
+    SENSOR_GAS: ['Gas Resistance', 'Ohms'],
+    SENSOR_AQ: ['Air Quality', '%']
+}
+DEFAULT_MONITORED = [SENSOR_TEMP, SENSOR_HUMID, SENSOR_PRESS, SENSOR_AQ]
+OVERSAMPLING_VALUES = set([0, 1, 2, 4, 8, 16])
+FILTER_VALUES = set([0, 1, 3, 7, 15, 31, 63, 127])
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_I2C_ADDRESS, default=DEFAULT_I2C_ADDRESS): cv.string,
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=DEFAULT_MONITORED):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Optional(CONF_I2C_BUS, default=DEFAULT_I2C_BUS): cv.positive_int,
+    vol.Optional(CONF_OVERSAMPLING_TEMP, default=DEFAULT_OVERSAMPLING_TEMP):
+        vol.All(vol.Coerce(int), vol.In(OVERSAMPLING_VALUES)),
+    vol.Optional(CONF_OVERSAMPLING_PRES, default=DEFAULT_OVERSAMPLING_PRES):
+        vol.All(vol.Coerce(int), vol.In(OVERSAMPLING_VALUES)),
+    vol.Optional(CONF_OVERSAMPLING_HUM, default=DEFAULT_OVERSAMPLING_HUM):
+        vol.All(vol.Coerce(int), vol.In(OVERSAMPLING_VALUES)),
+    vol.Optional(CONF_FILTER_SIZE, default=DEFAULT_FILTER_SIZE):
+        vol.All(vol.Coerce(int), vol.In(FILTER_VALUES)),
+    vol.Optional(CONF_GAS_HEATER_TEMP, default=DEFAULT_GAS_HEATER_TEMP):
+        vol.All(vol.Coerce(int), vol.Range(200, 400)),
+    vol.Optional(CONF_GAS_HEATER_DURATION,
+                 default=DEFAULT_GAS_HEATER_DURATION):
+        vol.All(vol.Coerce(int), vol.Range(1, 4032)),
+    vol.Optional(CONF_AQ_BURN_IN_TIME, default=DEFAULT_AQ_BURN_IN_TIME):
+        cv.positive_int,
+    vol.Optional(CONF_AQ_HUM_BASELINE, default=DEFAULT_AQ_HUM_BASELINE):
+        vol.All(vol.Coerce(int), vol.Range(1, 100)),
+    vol.Optional(CONF_AQ_HUM_WEIGHTING, default=DEFAULT_AQ_HUM_WEIGHTING):
+        vol.All(vol.Coerce(int), vol.Range(1, 100)),
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the BME680 sensor."""
+    SENSOR_TYPES[SENSOR_TEMP][1] = hass.config.units.temperature_unit
+    name = config.get(CONF_NAME)
+
+    sensor_handler = yield from hass.async_add_job(_setup_bme680, config)
+    if sensor_handler is None:
+        return False
+
+    dev = []
+    try:
+        for variable in config[CONF_MONITORED_CONDITIONS]:
+            dev.append(BME680Sensor(
+                sensor_handler, variable, SENSOR_TYPES[variable][1], name))
+    except KeyError:
```

Comment:
This can't happen with config validation in place. You can remove the try... except.